src/Heist/Extra/Splices/Pandoc/Ctx.hs

```diff
@@ -115,16 +121,75 @@ inlineLookupAttr node = \case
   B.Code {} -> childTagAttr node "Code"
   B.Note _ ->
     childTagAttr node "Note"
-  B.Link _ _ (url, _) ->
-    fromMaybe B.nullAttr $ do
-      link <- X.childElementTag "PandocLink" node
-      let innerTag = if "://" `T.isInfixOf` url then "External" else "Internal"
-      pure $ attrFromNode link `concatAttr` childTagAttr link innerTag
+  B.Link (_, classes, _) inlines (url, _) ->
+    linkLookupAttr node classes inlines url
   _ -> B.nullAttr
 
+linkLookupAttr :: X.Node -> [Text] -> [B.Inline] -> Text -> B.Attr
+linkLookupAttr node classes inlines url =
+  let linkType = determineLinkType inlines classes url
+   in fromMaybe B.nullAttr $ do
+        linkNode <- X.childElementTag "PandocLink" node
+        let linkTypeNode =
+              X.childElementTag
+                (case linkType of LinkType_Internal -> "Internal"; LinkType_External _ -> "External")
+                linkNode
+        let iconNode =
+              case linkType of
+                LinkType_Internal -> Nothing
+                LinkType_External i ->
+                  linkTypeNode
+                    >>= X.childElementTag
+                      (case i of Icon -> "Icon"; NoIcon -> "NoIcon")
+        pure $
+          attrFromNode linkNode
+            `concatAttr` attrFromMaybeNode linkTypeNode
+            `concatAttr` attrFromMaybeNode iconNode
+
+data LinkType = LinkType_Internal | LinkType_External ExternalLinkIconStatus
+  deriving stock (Eq, Show)
+
+data ExternalLinkIconStatus = Icon | NoIcon
+  deriving stock (Eq, Show)
```

Comment:
I'm curious why you went with two different types. Why not instead this:

```haskell
-- Regular links do not have any special styling (like icons)
data LinkType = LinkType_Regular | LinkType_Internal | LinkType_External
```

This way, I imagine users can explicitly choose them in CSS style using `[foo](mailto:s@s.com){data-linktype=regular)`.

Reply:
> I'm curious why you went with two different types. Why not instead this:

1. Because I know what an internal link is (a link pointing to an internal resource of the site I'm visiting) and what an external link is (a link pointing to an off-site resource). And I know what is an external link with an icon and an external link without an icon. But I have no clue about a regular (or a [normal](https://github.com/EmaApps/emanote/discussions/348#discussioncomment-3609432)) link. (And the name suggests these should be the prevalent ones). For me the icon thing is about the styling, not about the link type.
2. I believe I do have three types after all: Internal, External with an icon, External without an icon.
3. Because in `pandoc.tpl` I would have to repeat the `target="_blank" rel="noopener"` attributes for the *regular* links, and I didn't like the repetition:

```
  <PandocLink class="text-${theme}-600">
    <Internal class="mavenLinkBold hover:underline" />
    <External class="hover:underline" target="_blank" rel="noopener" />
  </PandocLink>
```

> This way, I imagine users can explicitly choose them in CSS style using [foo](mailto:s@s.com){data-linktype=regular).

I used the `data` idea.